src/main/java/com/daml/extensions/testing/utils/PackageUtils.java

```diff
@@ -199,6 +204,39 @@ public static DamlLf1.Package findPackageObject(DamlLedgerClient ledgerClient, S
         ledgerClient, DamlLf1.DottedName.newBuilder().addSegments(moduleName).build());
   }
 
+  public static TemplateType waitForTemplate(
+      DamlLedgerClient ledgerClient,
+      String moduleAndEntityName,
+      Duration waitTimeout,
+      Logger logger)
+      throws TimeoutException, InvalidProtocolBufferException {
+    boolean found = false;
+    Stopwatch time = Stopwatch.createStarted();
+    PackageUtils.TemplateType idWithArgs = null;
+    while (!found && time.elapsed().compareTo(waitTimeout) <= 0) {
+      try {
+        idWithArgs = findTemplate(ledgerClient, moduleAndEntityName);
+        found = true;
+      } catch (Exception ignored) {
+        if (!ignored.getMessage().contains("PACKAGE")) {
```

Comment:
Is this enough to know that we got the "right kind of exception" that we can ignore?